docs: add note for building on macOS with Apple Silicon (#30947)
camunda/camunda: CONTRIBUTING.md

  .
```

#### Build on macOS with an Apple Silicon chip

> > [!NOTE]
> > When building the code on a macOS with a Silicon chip(e.g. M1-M4), The build can fail with protoc errors.

Reviewer: 🔧 This comment will go out-of-date as soon as M5 comes out. We can avoid that by being less specific.

```suggestion
> [!NOTE]
> When building the code on a macOS with a Silicon chip (e.g. M1, M2, etc), the build can fail with protoc errors.
```

> If you are running into protoc issues make sure that you have Rosetta installed locally.

Reviewer: ```suggestion
If you are running into protoc issues, make sure that you have Rosetta installed locally.
```

Rosetta enables a Mac with Apple silicon to use apps that were built for a Mac with an Intel processor.

> If you don't have Rosetta installed locally , you can  install it with the following command :

Reviewer: ```suggestion
If you don't have Rosetta installed locally, you can  install it with the following command:
```

Author: thank you for the great feedback
